Mathlib/Data/Set/Image.lean

```diff
@@ -814,10 +814,26 @@ theorem compl_range_inl : (range (Sum.inl : α → α ⊕ β))ᶜ = range (Sum.i
 theorem compl_range_inr : (range (Sum.inr : β → α ⊕ β))ᶜ = range (Sum.inl : α → α ⊕ β) :=
   IsCompl.compl_eq isCompl_range_inl_range_inr.symm
 
+@[simp]
+theorem image_inl_inter_image_inr {X Y} (u : Set X) (v : Set Y) :
+    Sum.inl '' u ∩ Sum.inr '' v = ∅ := by
+  apply subset_empty_iff.mp
+  intro x hx
+  nomatch hx
+
+theorem preimage_sumElim_eq {f : α → γ} {g : β → γ} (S : Set γ) :
```

Comment:
Can you name the variable `s`, for consistency with most of this file (and mathlib style)? Thanks!